app/controllers/unattended_controller.rb

```diff
@@ -12,6 +12,7 @@ class UnattendedController < ApplicationController
     alias_method_chain f, :unattended
   end
 
+  before_action :set_admin_user, :unless => Proc.new { preview? }
```

Comment:
I think `preview?` will return false on hostgroup template view, leading to any user being able to possibly see details they should not.